server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/ProcessingContext.scala

```diff
@@ -45,10 +45,8 @@ case class PlainPart(name: String) extends JsonPathPart {
 
 object ArrayElementPart {
   def parse(string: String): Option[ArrayElementPart] = {
-    if (string.startsWith("[") && string.endsWith("]")) {
-      val positionPart: String = string.substring(1, string.length - 1)
-      Try(positionPart.toInt)
-        .fold(_ => None, fromInt)
+    if (string forall Character.isDigit) {
+      fromInt(string.toInt)
```

Comment:
We got read of some error handling here. Are we sure to handle all cases already?

Reply:
Yes, it has been thrown when "99999999999999999999999999"